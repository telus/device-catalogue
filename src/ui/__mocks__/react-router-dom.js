import React from "react";

const reactRouterDom = require("react-router-dom");
// Mock to avoid colission with MemoryRouter when testing.
// Just render plain div with its children
reactRouterDom.BrowserRouter = ({ children }) => <div>{children}</div>;

module.exports = reactRouterDom;
