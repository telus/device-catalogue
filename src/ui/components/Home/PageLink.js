import React from 'react';
import Link from '../Common/LinkComponent';
import TdsLink from '@tds/core-link';
import Card from '@tds/core-card';
import Box from '@tds/core-box';
import Image from '@tds/core-image';

const PageLink = ({ image, path, linkText, altText }) => {
  return (
    <Card variant="defaultOnlyBorder">
      <Box between={3} className="item-links">
        <Image src={image} alt={altText} width={90}/>
        <div>
          <Link TDSLink={TdsLink} to={path}>{linkText}</Link>
        </div>
      </Box>
    </Card>
  )
}

export default PageLink;