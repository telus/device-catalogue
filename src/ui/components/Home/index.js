import React from "react";
import { Link } from "react-router-dom";
import Heading from "@tds/core-heading";
import Box from "@tds/core-box";
import Paragraph from "@tds/core-paragraph";
import Image from "@tds/core-image";
import HairlineDivider from "@tds/core-hairline-divider";
import FlexGrid from "@tds/core-flex-grid";
import DimpleDivider from "@tds/core-dimple-divider";
import phoneImage from "../../../../public/images/iphone.jpg";
import watchImage from "../../../../public/images/watch.jpg";
import telusImage from "../../../../public/images/TELUS_TAGLINE_HORIZONTAL_EN.svg";
import styled from "styled-components";

const CenteredText = styled.span`
  text-align: center;
  font-weight: 600;
  font-size: 1.1rem;
  text-decoration: none;
`;

const HoverEffectContainer = styled(Box)`
  img {
    transform: scale(1);
    transition: 0.5s ease-in-out;
    margin-bottom: 20px;
  }

  &:hover img {
    transform: scale(1.1);
  }
`;

const DeviceLink = styled(Link)`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const Home = () => {
  return (
    <>
      <FlexGrid>
        <FlexGrid.Row>
          <FlexGrid.Col xs={12}>
            <Box between={3} vertical={4}>
              <Image src={telusImage} width={150} height={50} alt="TELUS" />
              <HairlineDivider />
              <Heading level="h1">Device Catalog</Heading>
              <Paragraph>Revolutionary devices at unbeatable prices.</Paragraph>
              <DimpleDivider />
            </Box>
          </FlexGrid.Col>
        </FlexGrid.Row>
        <FlexGrid.Row>
          <FlexGrid.Col xs={12}>
            <Box between={4}>
              <Box inline between={5} below={4}>
                <HoverEffectContainer between={3} horizontal={4}>
                  <DeviceLink to="/iphones">
                    <Image
                      src={phoneImage}
                      alt="iPhone image"
                      width={150}
                      height={250}
                    />
                    <CenteredText>iPhones</CenteredText>
                  </DeviceLink>
                </HoverEffectContainer>
                <HoverEffectContainer between={3} inline={false}>
                  <DeviceLink to="/watches">
                    <Image
                      src={watchImage}
                      alt="Watch image"
                      width={150}
                      height={250}
                    />
                    <CenteredText>Watches</CenteredText>
                  </DeviceLink>
                </HoverEffectContainer>
              </Box>
              <HairlineDivider />
            </Box>
          </FlexGrid.Col>
        </FlexGrid.Row>
      </FlexGrid>
    </>
  );
};

export default Home;
