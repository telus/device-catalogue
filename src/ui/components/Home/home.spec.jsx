import React from "react";
import { render } from "enzyme";
import Home from ".";
import { MemoryRouter } from "react-router-dom";

describe("Device Catalog", () => {
  it("renders correctly", () => {
    const container = render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );

    expect(container.html()).toMatchSnapshot();
    expect(container.find("img").length).toEqual(3);
    expect(container.html().includes("Device Catalog")).toEqual(true);
    expect(container.html().includes("iPhones")).toEqual(true);
    expect(container.find('a[href="/iphones"]').length).toEqual(1);
    expect(container.html().includes("Watches")).toEqual(true);
    expect(container.find('a[href="/watches"]').length).toEqual(1);
  });
});
