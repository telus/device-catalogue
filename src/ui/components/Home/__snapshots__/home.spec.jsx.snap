// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Device Catalog renders correctly 1`] = `"<div class=\\"Row__StyledRow-sc-1goz0ht-0 fObXZQ\\"><div class=\\"Col__StyledCol-sc-15yvjc7-0 gBwSfh\\"><div class=\\"sc-bdvvtL cluyzB\\"><img src=\\"../public/images/TELUS_TAGLINE_HORIZONTAL_EN.svg\\" width=\\"150\\" height=\\"50\\" alt=\\"TELUS\\" class=\\"Image__StyledImage-b16p20-0 cNmOoM\\"><hr class=\\"HairlineDivider__StyledHairlineDivider-biamlk-0 BvZCb\\"><h1 data-testid=\\"heading\\" class=\\"Heading__StyledHeading-mpz92r-0 fflItf\\">Device Catalog</h1><p class=\\"Paragraph__StyledParagraph-efl81j-0 jukEfT\\">Revolutionary devices at unbeatable prices.</p><hr class=\\"DimpleDivider__StyledDimpleDivider-quxnvh-0 ibPxnT\\"></div></div></div><div class=\\"Row__StyledRow-sc-1goz0ht-0 fObXZQ\\"><div class=\\"Col__StyledCol-sc-15yvjc7-0 gBwSfh\\"><div class=\\"sc-bdvvtL dGyuLk\\"><div class=\\"sc-bdvvtL YWUYw\\"><div class=\\"sc-bdvvtL cdVIWp sc-eCImPb iUidrU sc-eCImPb iUidrU\\"><a class=\\"sc-jRQBWg cxlYPA\\" href=\\"/iphones\\"><img src=\\"../public/images/iphone.jpg\\" width=\\"150\\" height=\\"250\\" alt=\\"iPhone image\\" class=\\"Image__StyledImage-b16p20-0 cNmOoM\\"><span class=\\"sc-hKwDye jeLaXE\\">iPhones</span></a></div><div class=\\"sc-bdvvtL iQfmHk sc-eCImPb iUidrU sc-eCImPb iUidrU\\"><a class=\\"sc-jRQBWg cxlYPA\\" href=\\"/watches\\"><img src=\\"../public/images/watch.jpg\\" width=\\"150\\" height=\\"250\\" alt=\\"Watch image\\" class=\\"Image__StyledImage-b16p20-0 cNmOoM\\"><span class=\\"sc-hKwDye jeLaXE\\">Watches</span></a></div></div><hr class=\\"HairlineDivider__StyledHairlineDivider-biamlk-0 BvZCb\\"></div></div></div>"`;
